### PR TITLE
CTCP-4961: additionalReference yes no page

### PR DESCRIPTION
**Pull request by AdnanGondal**
image

### CODE REVIEW

File: app/pages/additionalReference/AdditionalReferenceNumberPage.scala [hunk 1]
```diff
@@ -16,11 +16,13 @@
 
 package pages.additionalReference
 
-import models.Index
```
Comment (JakeGoodwin) [inline app/pages/additionalReference/AdditionalReferenceNumberPage.scala:19]: unused imports

File: app/pages/additionalReference/AdditionalReferenceNumberPage.scala [hunk 2]
```diff
@@ -16,11 +16,13 @@
...
-import pages.QuestionPage
+import models.{Index, UserAnswers}
+import pages.{QuestionPage, UnloadingCommentsPage}
```
Comment (justega247) [inline app/pages/additionalReference/AdditionalReferenceNumberPage.scala:20]: Quick one here, can you remove these unused imports?

File: app/pages/additionalReference/AdditionalReferenceNumberPage.scala [hunk 3]
```diff
@@ -16,11 +16,13 @@
...
 import pages.sections.additionalReference.AdditionalReferenceSection
 import play.api.libs.json.JsPath
 
+import scala.util.Try
```
Comment (justega247) [inline app/pages/additionalReference/AdditionalReferenceNumberPage.scala:24]: Same as above

File: app/pages/additionalReference/AdditionalReferenceNumberYesNoPage.scala [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package pages.additionalReference
+
+import models.{Index, UserAnswers}
+import pages.{QuestionPage}
```
Comment (JakeGoodwin) [inline app/pages/additionalReference/AdditionalReferenceNumberYesNoPage.scala:20]: We don't need the braces here

File: test/pages/additionalReference/AdditionalReferenceNumberYesNoPageSpec.scala [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+            val result = userAnswers.setValue(AdditionalReferenceNumberYesNoPage(index), false)
+
+            result.get(AdditionalReferenceNumberPage(index)) must not be defined
+
+        }
+      }
+
+      "must keep additional reference number when yes selected" in {
+        forAll(Gen.alphaNumStr) {
```
Comment (justega247) [inline test/pages/additionalReference/AdditionalReferenceNumberYesNoPageSpec.scala:48]: Same as above

**Review by JakeGoodwin — APPROVED** [review]

**Pull request merged**